.travis.yml

```diff
@@ -1,35 +1,62 @@
-addons:
-  apt:
-    packages:
-        - wget
-        - locales
-        - python3-pip
+  # Testing QuickLogic toolchain on all OSes
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=trusty
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=xenial
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=bionic
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=eoan
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=focal
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=6
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=7
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=8
+  # Testing Xilinx 7 Series toolchain on all OSes
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=trusty
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=xenial
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=bionic
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=eoan
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=focal
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=6
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=7
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=8
 
 before_install:
         - sudo locale-gen "en_US.UTF-8"
         - export LANG="en_US.UTF-8"
         - source .github/travis/common.sh
         - rm README.rst && make README.rst && .github/travis/git-check.sh
-        - pip install git+https://github.com/antmicro/tuttest
-        - tuttest README.rst unnamed0 | bash -
-jobs:
-  include:
-    - stage: SymbiFlow
-      install:
-            - tuttest xc7/README.rst unnamed0 | bash -
+        - pip install git+https://github.com/antmicro/tuttest@prefix-lines-with#egg=tuttest
 
-      script:
-            - tuttest xc7/README.rst unnamed1,unnamed2,unnamed3,unnamed4,unnamed5 | bash -
-    - #
-      install:
-            - tuttest eos-s3/README.rst unnamed0 | bash -
+install:
+ - if [ "$OS" = "ubuntu" ]; then export INSTALL_COMMAND="apt update && apt install -y"; fi
+ - if [ "$OS" = "centos" ]; then export INSTALL_COMMAND="yum -y install"; fi
+ - if [ "$OS" = "ubuntu" ]; then export INSTALL_PACKAGES="wget"; fi
+ - if [ "$OS" = "centos" ]; then export INSTALL_PACKAGES="wget"; fi
+ - if [ "$OS" = "centos" ] && [ "$OS_VERSION" = "6" ]; then export INSTALL_PACKAGES="$INSTALL_PACKAGES util-linux-ng"; fi # Install missing `getopt` binary in centos 6
+ - export DOCKER_NAME=test
+ - export IN_DOCKER_EXEC="docker exec -t $DOCKER_NAME bash -c"
+ - docker create
+        --name $DOCKER_NAME
+        --env LANG="en_US.UTF-8"
+        --env LANGUAGE="en_US"
+        --env DEBIAN_FRONTEND="noninteractive"
+        -v /home/travis:/home/travis
+        --workdir $PWD
+        --tty
+        $OS:$OS_VERSION
+ - docker start $DOCKER_NAME
+ - $IN_DOCKER_EXEC "$INSTALL_COMMAND $INSTALL_PACKAGES"
+ - if [ "$OS_VERSION" = "6" ] || [ "$OS_VERSION" = "trusty" ]; then $IN_DOCKER_EXEC "echo 'readlink -f \$@' | tee -a /usr/bin/realpath;chmod +x /usr/bin/realpath"; fi # Replace missing `realpath` binary with readlink -f
```

Comment:
This line seems a bit weird....

Reply:
Quicklogic toolchain wrappers refers to the files, that are no longer shipped with the toolchain. [realpath](https://launchpad.net/ubuntu/trusty/+package/realpath) package from Ubuntu 14 returns an error code when it tries to resolve path to file that do not exists. 

``readlink -f`` is very close to ``realpath``, the only difference is that it doesn't emulate ``realpath`` rarely-used command line options.

I created PR to remove this references to files from toolchain wrappers: https://github.com/QuickLogic-Corp/symbiflow-arch-defs/pull/161

Reply:
I updated quicklogic conda environment file to install ``realpath`` binary inside conda env.
This version of ``realpath`` do not returns an error code and it allows to do not use ``readlink -f```